lib/src/signed_video_h26x_sign.c

```diff
@@ -603,12 +592,12 @@ signed_video_get_lastest_sei(signed_video_t *self, uint8_t *sei, size_t *sei_siz
   // Assign the SEI size to the provided pointer.
   *sei_size = self->sei_data_buffer[self->num_of_completed_seis - 1].completed_sei_size;
   if (!sei) return SV_OK;
-  // Assign SEI data to the provided pointer.
+  // Copy SEI data to the provided pointer.
   memcpy(sei, self->sei_data_buffer[self->num_of_completed_seis - 1].payload, *sei_size);
 
   // Reset the fetched SEI information from the sei buffer.
   --(self->num_of_completed_seis);
-  reset_sei_buffer_at_index(self, self->num_of_completed_seis);
+  shift_sei_buffer_at_index(self, self->num_of_completed_seis);
```

Comment:
You will leak memory on this line. Run valgrind to see if there are other memory leaks